Smarty/Smarty.class.php

```diff
@@ -1543,7 +1543,8 @@ function validateCompilePath( $compilePath ) {
         }
 
         $isValid = false;
-        if ( $fd = @fopen( $compilePath, 'wb') ) {
+        // Open file for writing but do not truncate.
+        if ( $fd = @fopen( $compilePath, 'c') ) {
             $isValid = true;
             @fclose( $fd );
             @unlink($compilePath);
```

Comment:
This isn't an issue with your patch, but a surprising observation about this section of code -- it looks like `validateCompilePath()` (*"do check can smarty create a file w/ given path" [sic]*) would effectively disable caching. If I'm reading correctly, every call to `Smarty::fetch()` calls `Smarty::_get_compile_path()`; as long as the resource uses `string:` notation, it will then call  `Smarty::validateCompilePath()` which then `@fopen`s and `@unlink`s the file... which means that cache file is always deleted before you get a chance to use it.

(Hrm... this whole scenario for caching `string:` seems overwrought...  as in... it's not sure what to name the cache-file, so it does `@fopen+@fclose+@unlink` to see if the filename produces a syntax error; and if there is an error, then it makes a uniqueish-but-not-reusable name with `time()+rand()`. Wouldn't it be simpler to have a more robust naming pattern like `$this->compile_dir . sha1($resource_name) . '.php'`?)